olive/passes/onnx/graph_surgeries.py

```diff
@@ -147,28 +126,29 @@ class RemoveInitializerFromInputs(Surgeon):
     def __init__(self):
         pass
 
-    def __call__(self, model: ModelProto):
-        initializer_names = {initializer.name for initializer in model.graph.initializer}
-        updated_inputs = [graph_input for graph_input in model.graph.input if graph_input.name not in initializer_names]
-        del model.graph.input[:]
-        model.graph.input.extend(updated_inputs)
+    def call_ir(self, model: ir.Model) -> ir.Model:
+        while model.graph.inputs[-1].name in model.graph.initializers:
```

Comment:
Nit: may not be a real case like all inputs are initializer then this will get index errors. Maybe some testing models can have this case?

Reply:
That’s a valid point. Will check size first

Reply:
Done